src/views/CardStack.js

```diff
@@ -181,41 +164,24 @@ class CardStack extends Component {
     }
   }
 
-  _reset(resetToIndex: number, velocity: number): void {
+  _reset(resetToIndex: number): void {
     Animated.timing(this.props.position, {
         toValue: resetToIndex,
         duration: ANIMATION_DURATION,
         useNativeDriver: this.props.position.__isNative,
-        velocity: velocity * GESTURE_ANIMATED_VELOCITY_RATIO,
-        bounciness: 0,
       })
       .start();
   }
 
-  _goBack(backFromIndex: number, velocity: number) {
-    const {navigation, position, scenes} = this.props;
+  _goBack(backFromIndex: number) {
+    const {navigation, scenes} = this.props;
     const toValue = Math.max(backFromIndex - 1, 0);
-
-    // set temporary index for gesture handler to respect until the action is
-    // dispatched at the end of the transition.
-    this._immediateIndex = toValue;
-
-    Animated.timing(position, {
-        toValue,
-        duration: ANIMATION_DURATION,
-        useNativeDriver: position.__isNative,
-        velocity: velocity * GESTURE_ANIMATED_VELOCITY_RATIO,
-        bounciness: 0,
-      })
-      .start(({finished}) => {
-        this._immediateIndex = null;
-        const backFromScene = scenes.find(s => s.index === toValue + 1);
-        if (!this._isResponding && backFromScene) {
-          navigation.dispatch(
-            NavigationActions.back({ key: backFromScene.route.key })
-          );
-        }
-      });
+    const backFromScene = scenes.find(s => s.index === toValue + 1);
+    if (!this._isResponding && backFromScene) {
+      navigation.dispatch(
```

Comment:
This dispatches an action in the middle of the gesture! This causes us to loose a bunch of frames at the time when the touch is released. That is why this optimization was made. We wait to fire the back action until the gesture actually completes.